subsys/net/openthread/rpc/common/ot_rpc_common.c

```diff
@@ -147,3 +149,111 @@ void ot_rpc_report_decoding_error(uint8_t cmd_evt_id)
 {
 	nrf_rpc_err(-EBADMSG, NRF_RPC_ERR_SRC_RECV, &ot_group, cmd_evt_id, NRF_RPC_PACKET_TYPE_CMD);
 }
+
+bool ot_rpc_encode_message_info(struct nrf_rpc_cbor_ctx *ctx, const otMessageInfo *aMessageInfo)
+{
+	uint8_t tmp;
+
+	if (!zcbor_bstr_encode_ptr(ctx->zs, (const char *)&aMessageInfo->mSockAddr.mFields.m8,
+				   sizeof(aMessageInfo->mSockAddr))) {
+		return false;
+	}
+
+	if (!zcbor_bstr_encode_ptr(ctx->zs, (const char *)&aMessageInfo->mPeerAddr.mFields.m8,
+				   sizeof(aMessageInfo->mPeerAddr))) {
+		return false;
+	}
+
+	if (!zcbor_uint_encode(ctx->zs, &aMessageInfo->mSockPort,
+			       sizeof(aMessageInfo->mSockPort))) {
+		return false;
+	}
+
+	if (!zcbor_uint_encode(ctx->zs, &aMessageInfo->mPeerPort,
+			       sizeof(aMessageInfo->mPeerPort))) {
+		return false;
+	}
+
+	if (!zcbor_uint_encode(ctx->zs, &aMessageInfo->mHopLimit,
+			       sizeof(aMessageInfo->mHopLimit))) {
+		return false;
+	}
+
+	tmp = aMessageInfo->mEcn;
+	if (!zcbor_uint_encode(ctx->zs, &tmp, sizeof(tmp))) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, aMessageInfo->mIsHostInterface)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, aMessageInfo->mAllowZeroHopLimit)) {
+		return false;
+	}
+
+	if (!zcbor_bool_put(ctx->zs, aMessageInfo->mMulticastLoop)) {
+		return false;
+	}
+
+	return true;
+}
+
+bool ot_rpc_decode_message_info(struct nrf_rpc_cbor_ctx *ctx, otMessageInfo *aMessageInfo)
+{
+	struct zcbor_string zst;
+	uint8_t tmp;
+
+	if (!zcbor_bstr_decode(ctx->zs, &zst)) {
```

Comment:
Again, it is more optimal to use nrf rpc deserializer to not check the result of every decoding.